pkg/common/cluster/healthchecks/pods.go

```diff
@@ -79,12 +78,48 @@ func checkPods(podClient v1.CoreV1Interface, logger *log.Logger, filters ...PodP
 
 	logger.Printf("%v pods are currently not running or complete:", len(pods.Items))
 
+	// Create a map of pods with their cronJob names under the given namespace. If pod with no associated cronJob is not Pending, simply return the error message.
+	filteredPods := map[string]map[string]interface{}{}
+	namespace := ""
+	jobName := 0
+	cronJobName := ""
 	for _, pod := range pods.Items {
-		if pod.Status.Phase != kubev1.PodPending {
-			return nil, fmt.Errorf("Pod %s in unexpected phase %s: reason: %s message: %s", pod.GetName(), pod.Status.Phase, pod.Status.Reason, pod.Status.Message)
+		if pod.ObjectMeta.Labels["job-name"] == "" {
+			if pod.Status.Phase != kubev1.PodPending {
+				return nil, fmt.Errorf("Pod %s in unexpected phase %s: reason: %s message: %s", pod.GetName(), pod.Status.Phase, pod.Status.Reason, pod.Status.Message)
+			}
+			logger.Printf("%s is not ready. Phase: %s, Message: %s, Reason: %s", pod.Name, pod.Status.Phase, pod.Status.Message, pod.Status.Reason)
+		}
+		if namespace == "" || namespace != pod.Namespace {
+			filteredPods[pod.Namespace] = map[string]interface{}{}
+		}
+		jobName = strings.LastIndex(pod.ObjectMeta.Labels["job-name"], "-")
```

Comment:
It would be good just for safety to check whether this has returned `-1` or not (meaning the `LastIndex` didn't find anything) and return an error, because otherwise it will cause a panic when used in the subsequent lines.

Reply:
Isn't content of labels already checked on the line 87? Though I agree that specifically 'strings.LastIndex' is not.

Reply:
I have added a condition that checks if `LastIndex` returned -1 which creates the safety net.

Reply:
Thanks a bunch for that!